azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go

```diff
@@ -312,98 +355,143 @@ func resourceMonitorActivityLogAlertDelete(d *schema.ResourceData, meta interfac
 			Field:  utils.String("resourceId"),
 			Equals: utils.String(id),
 		})
 	}
 	if status := v["status"].(string); status != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("status"),
 			Equals: utils.String(status),
 		})
 	}
 	if subStatus := v["sub_status"].(string); subStatus != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("subStatus"),
 			Equals: utils.String(subStatus),
 		})
 	}
 	if recommendationType := v["recommendation_type"].(string); recommendationType != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationType"),
 			Equals: utils.String(recommendationType),
 		})
 	}
 
 	if recommendationCategory := v["recommendation_category"].(string); recommendationCategory != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationCategory"),
 			Equals: utils.String(recommendationCategory),
 		})
 	}
 
 	if recommendationImpact := v["recommendation_impact"].(string); recommendationImpact != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationImpact"),
 			Equals: utils.String(recommendationImpact),
 		})
 	}
 
-	return &insights.ActivityLogAlertAllOfCondition{
+	if serviceHealth := v["servicehealth"].([]interface{}); len(serviceHealth) > 0 {
+		for _, serviceItem := range serviceHealth {
+			vs := serviceItem.(map[string]interface{})
+			if regions, ok := vs["regions"]; ok {
+				if rv, ok := regions.(*schema.Set); ok {
+					if len(rv.List()) > 0 {
+						conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
+							Field:       utils.String("properties.impactedServices[*].ImpactedRegions[*].RegionName"),
+							ContainsAny: utils.ExpandStringSlice(rv.List()),
+						})
+					}
+				}
+			}
+
+			if events, ok := vs["events"]; ok {
+				if ev, ok := events.(*schema.Set); ok {
+					if len(ev.List()) > 0 {
+						ruleLeafCondition := make([]insights.AlertRuleLeafCondition, 0)
+						for _, e := range ev.List() {
+							event := e.(string)
+							ruleLeafCondition = append(ruleLeafCondition, insights.AlertRuleLeafCondition{
+								Field:  utils.String("properties.incidentType"),
+								Equals: utils.String(event),
+							})
+						}
+						conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
+							AnyOf: &ruleLeafCondition,
+						})
+					}
+				}
```

Comment:
Could we please extract this to an expand function?

Reply:
Yes, sure